sympy/printing/c.py

```diff
@@ -628,6 +629,45 @@ def _print_BreakToken(self, _):
 
     def _print_ContinueToken(self, _):
         return 'continue'
+    
+    def _codeblock_Sum(self, expr, **kwargs):
+        # Connect to the rest of the expression
+        sum_result = CodeblockResult(expr, 'sumResult') 
+        sum_expr = Dummy('sumExpr')
+        sum_index = Dummy('sumIndex')
+        sum_start = Dummy('sumStart')
+        sum_end = Dummy('sumEnd')
+
+        # Replace the index term with the dummy index
+        expr = expr.xreplace({expr.args[1][0]: sum_index})
+
+        # Resolve the possible code blocks in the summand 
+        summand, summand_codeblocks = self.doblocks(expr.args[0])
```

Comment:
In the printers, recursive calls use `_print` instead of `doprint`. doprint is only used for the top-level user call to the printer. This is done so that doprint can do pre- and post-processing on the full expression. We would probably want to do the same thing here as well.

Reply:
Agree